### PR TITLE
Add tekton 0.20 api support and update the corresponding go client packages

### LINKED ISSUE #402: Upgrade API server to use Tekton version >= 0.19
/kind feature

**Description:**
[A clear and concise description of what your proposal. What problem does it solve?]
Since Tekton version v0.19.0 released, it contains the features (e.g custom task) we may need to use. But seems it was blocked by the issue of:
https://github.com/argoproj/argo/issues/4425

And the PR provided:
https://github.com/argoproj/argo/pull/4810

After that, we can upgrade kfp-tekton API Server to Tekton version 0.19.0.

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #402 

**Description of your changes:**
1. Remove Argo go package
2. Update client-go and the corresponding code gen to v0.18.12
3. Update Tekton client to 0.20.1 (Support custom task with workspaces and pod template. Dynamic parameter support for custom task is coming in 0.21.x release)
4. Regenerate CRD client and update all client-go related functions to pass context and k8s options.

To test this deployment, run `make build-backend-images` or follow the [developer guide](https://github.com/kubeflow/kfp-tekton/blob/master/guides/developer_guide.md#backend) to run it with your test images.

Here is my prebuilt images if someone can't install golang. 
```
aipipeline/api-server:tekton-0.20
aipipeline/persistenceagent:tekton-0.20
aipipeline/scheduledworkflow:tekton-0.20
```

**Environment tested:**

* Python Version (use `python --version`): 3.7
* Tekton Version (use `tkn version`): 0.20.1
* Kubernetes Version (use `kubectl version`): 1.18
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: guides/developer_guide.md [hunk 1]
```diff
@@ -78,8 +78,8 @@ npm run docker
 ```
 
 ### Backend
-The KFP backend with Tekton uses a modified version of Kubeflow Pipelines api-server, persistent agent, and metadata writer. 
-1. To build these two images, clone this repository under the [GOPATH](https://golang.org/doc/gopath_code.html#GOPATH) and rename it to `pipelines`. 
+The KFP backend with Tekton uses a modified version of Kubeflow Pipelines api-server, persistent agent, and metadata writer.
+1. To build these two images, clone this repository under the [GOPATH](https://golang.org/doc/gopath_code.html#GOPATH) and rename it to `pipelines`.
```
Comment (fenglixa) [inline guides/developer_guide.md:82]: From my testing, seems we don't need to rename it to `pipelines`, could you double confirm? Thanks

**Comment by fenglixa** [comment]
Thanks @Tomcli , It's really great! Overall, it's good to me.

File: guides/developer_guide.md (hunk 1 above)
Comment (Tomcli) [reply guides/developer_guide.md:82]: For building the docker image, the repo name doesn't matter. For building the go binaries, it depends on the user's GOPATH. Some users may see errors if their gopath is not matching with the directory/package path.

**Comment by fenglixa** [comment]
Thanks @Tomcli 
/lgtm

**Comment by ckadner** [comment]
did we want to create a `0.6.0` release before merging this PR?

**Pull request merged**